### ISSUE TITLE
Allow environment variables to reference shared environment variables

### ISSUE THREAD

**Issue by ariehf** [Huachao/vscode-restclient#342]
Allow environment variables to reference shared environment variables.
This is useful when there are a large amount of environments.
(can choose to be implemented with or without using the "$shared " prefix)

For example multiple environments where half of the environments use one shared variable, and the other half uses a different shared variable.

    "rest-client.environmentVariables": {
        "$shared": {
            "tokenDev": "Bearer xyz...",
            "tokenProd": "Bearer abc..."
        },
        "webserverDev1": {
            "token": "{{$shared tokenDev}}",
            "host": "http://webserverDev1",
        },
        "webserverDev2": {
            "token": "{{$shared tokenDev}}",
            "host": "http://webserverDev2",
        },
        "webserverProd1": {
            "token": "{{$shared tokenProd}}",
            "host": "http://webserverProd1",
        },
        "webserverProd2": {
            "token": "{{$shared tokenProd}}",
            "host": "http://webserverProd2",
        }
    }

GET {{host}}/myendpoint
Authorization: {{token}}

**Comment by uriel-sf** [comment]
👍

**Comment by uriel-sf** [comment]
I have a related use case. I'd like to reference different shared variables for different environments. For example:
```
{
  "rest-client.environmentVariables": {
    "$shared": {
        "devApiUrl": "dev URL",
        "qaApiUrl": "qa URL",
    },
    "dev.user.multirate": {
      "apiUrl": "{{$shared devApiUrl}}",
    },
    "dev.user.perpetual": {
      "apiUrl": "{{$shared devApiUrl}}",
    },
    "qa.user.multirate": {
         "apiUrl": "{{$shared qaApiUrl}}"
     }
```

**Comment by snackb** [comment]
Mind if I take a look at this?

**Comment by Huachao** [comment]
@ariehf @uriel-sf @snackb with the great PR from @snackb this feature will be published in next release.

**Comment by Huachao** [comment]
@ariehf @uriel-sf @snackb you can try this feature implemented by @snackb in the latest version 0.2.1
